keps/sig-cli/3638-built-in-command-shadowing/README.md

```diff
@@ -0,0 +1,681 @@
+- [Alternatives](#alternatives)
+- [Infrastructure Needed (Optional)](#infrastructure-needed-optional)
+<!-- /toc -->
+
+## Release Signoff Checklist
+
+Items marked with (R) are required *prior to targeting to a milestone / release*.
+
+- [ ] (R) Enhancement issue in release milestone, which links to KEP dir in [kubernetes/enhancements] (not the initial KEP PR)
+- [ ] (R) KEP approvers have approved the KEP status as `implementable`
+- [ ] (R) Design details are appropriately documented
+- [ ] (R) Test plan is in place, giving consideration to SIG Architecture and SIG Testing input (including test refactors)
+  - [ ] e2e Tests for all Beta API Operations (endpoints)
+  - [ ] (R) Ensure GA e2e tests meet requirements for [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+  - [ ] (R) Minimum Two Week Window for GA e2e tests to prove flake free
+- [ ] (R) Graduation criteria is in place
+  - [ ] (R) [all GA Endpoints](https://github.com/kubernetes/community/pull/1806) must be hit by [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+This KEP enables external plugins being used as subcommands of built-in `create` command via new `--use-plugin` flag.
+
+## Motivation
+
+There is a high demand for enriching some built-in kubectl commands with new subcommands. More specifically take an instance of create command, 
+even though it supports the creation of 18 resources(as of v1.25) with one line command execution(e.g. kubectl create job), 
+new requests to support other resources as well are occasionally filed. 
+These requests are understandable given the proliferation of CustomResourceDefinitions, but is not sustainable from the SIG CLI maintainers point of view. 
+Because that might bring about not only increase the size of the binary but also create a maintenance burden.
+
+This KEP proposes a viable alternative to users via plugins instead of creating static subcommands inside kubectl built-in commands.
+
+### Goals
+
+* Enable external plugin usage for `kubectl create` command
+
+### Non-Goals
+
+* Implement a plugin
+
+## Proposal
+
+This KEP introduces a new boolean flag, namely `--use-plugin`. Without this flag, plugin mechanism will work
```

Comment:
I'd suggest not using flag and env variable, but only keep the env var for alpha and beta. Once this mechanism is fully stable, it'll be fully acceptable to shadow some sub-commands, but not all. Iow. you'll be able to shadow `kubectl create foo` because such built-in does not exist, but not `kubectl create job`. That mechanism should fill in the gap the users are requesting, namely supporting CRDs, and resources that are not implemented currently in `kubectl`. This has a limitation of not allowing to shadow everything, but I'd suggest maintaining similar restrain as we did when we initially said in [the existing plugin design](https://github.com/kubernetes/enhancements/tree/master/keps/sig-cli/2379-kubectl-plugins#limitations-of-the-existing-design).